lib/improver/tests/ensemble_copula_coupling/ensemble_copula_coupling/test_GeneratePercentilesFromMeanAndVariance.py

```diff
@@ -344,11 +344,62 @@ def test_number_of_percentiles(self):
         plugin = Plugin()
         result = plugin.process(
             current_forecast_predictor, current_forecast_variance,
-            no_of_percentiles)
+            no_of_percentiles=no_of_percentiles)
         self.assertEqual(
             len(raw_forecast.coord("realization").points),
             len(result.coord("percentile_over_realization").points))
 
+    @ManageWarnings(
+        ignored_messages=["Only a single cube so no differences",
+                          "Collapsing a non-contiguous coordinate."])
+    def test_list_of_percentiles(self):
+        """
+        Test that the plugin returns a cube with the expected percentiles
+        when a specific list of percentiles is provided.
+        """
+        cube = self.current_temperature_forecast_cube
+        current_forecast_predictor = cube.collapsed(
+            "realization", iris.analysis.MEAN)
+        current_forecast_variance = cube.collapsed(
+            "realization", iris.analysis.VARIANCE)
+
+        percentiles = [10, 25, 50, 75, 90]
+
+        plugin = Plugin()
+        result = plugin.process(
+            current_forecast_predictor, current_forecast_variance,
+            percentiles=percentiles)
+        self.assertEqual(
```

Comment:
You're not actually testing whether the data in the cube is correct. Should you?

Reply:
Added check of the data in the unit tests.